streams/src/main/java/org/apache/kafka/streams/kstream/ForeachProcessor.java

```diff
@@ -19,6 +19,10 @@
 import org.apache.kafka.streams.processor.api.Processor;
 import org.apache.kafka.streams.processor.api.Record;
 
+/**
+ * @deprecated Since 4.0. Use {@link org.apache.kafka.streams.kstream.internals.ForeachProcessor} instead.
```

Comment:
Indeed, this class should be only for internal usage. I'll remove the description `Use {@link org.apache.kafka.streams.kstream.internals.ForeachProcessor} instead.`, thanks.

Reply:
This should till be a JavDoc -- not just a in-line comment.
```
/**
 * @deprecates since 4.0 and should not be used any longer.
 */
```
Or something like this.

Reply:
@mjsax you are right. will file a follow-up

Reply:
#17156